plugins/rikki/heroeslounge/models/Match.php

```diff
@@ -148,6 +148,14 @@ public function belongsToSeason($season)
             || ($this->division != null && $this->division->playoff != null && $this->division->playoff->season != null && $this->division->playoff->season->id == $season->id));
     }
 
+    public function associatedSeasons()
```

Comment:
I'm a bit confused, a match can only belong to one season at most.

- It belongs to a division, which can belong to a season.
- It belongs to a division, which can belong to a playoff, which can belong to a season.
- It belongs to a playoff, which can belong to a season.

I could be mistaken in the syntax, but it seems like you're returning an array of options, instead of the singular season it belongs to or null?

Reply:
I thought that too, but this is adapted from the method above it.

The point is that there are 3 different ways for a season to be associated with a match and I see no guarantees that they will all be the same.  If we just had one object to look at, like division and that had the season every time (if present), then that would be easy.  But the models do not appear to be that clean currently.

Reply:
Said another way: if only one of these has a season, then it's fine.  And if they all have the same season, then it's still fine.

Reply:
It feels like you'd often be doing a lot of unneeded data retrieval, since a majority of our matches have a division season relation. I'm not sure how "expensive" it is to retrieve relational data, but I feel like a conditional would be more fitting to exit early.

Reply:
True enough, if it is confirmed that the precedence order from the method above applies here as well, I can just return the first season encountered.

Reply:
Do that, and also name the method getSeasonAttribute()
That way, you can just access it with $match->season.

Also please refactor Match::belongsToSeason to use this method